src/app/qgsmaptoolselectionhandler.cpp

```diff
@@ -263,9 +263,20 @@ void QgsMapToolSelectionHandler::selectPolygonPressEvent( QgsMapMouseEvent *e )
         auto vectorLayer = static_cast<QgsVectorLayer *>( layer );
         if ( vectorLayer->geometryType() == QgsWkbTypes::PolygonGeometry )
         {
+          QgsRectangle rect( x - sr, y - sr, x + sr, y + sr );
+          QgsCoordinateTransform transform = mCanvas->mapSettings().layerTransform( vectorLayer );
+
+          try
+          {
+            rect = transform.transformBoundingBox( rect, QgsCoordinateTransform::ReverseTransform );
+          }
+          catch ( QgsCsException &exception )
+          {
+            QgsDebugMsg( QStringLiteral( "Could not transform geometry to layer CRS" ) );
+          }
```

Comment:
This shouldn't be required -- when setDestinationCrs is called on a feature request then the filter rect transformation is handled automatically

Reply:
yes, but here I want the result feature request in layer coordinate, so I don not use setDestinationCrs.

Reply:
@nyalldawson is there still a problem here?

Reply:
I must admit I still don't follow the new logic -- I can't see anywhere in QgsIdentifyMenu where the crs is important, so I can't see why it's a problem to do the transformation once upfront.

I'm not doubting that this fixes the issue... but if I can't understand the code then it's a good sign that either API needs to be reworked or that we're missing a whole bunch of explanatory comments here (preferable the first!)